src/Compiler/Checking/CheckComputationExpressions.fs

```diff
@@ -3454,13 +3454,6 @@ let TcSequenceExpressionEntry (cenv: cenv) env (overallTy: OverallTy) tpenv (has
         let validateObjectSequenceOrRecordExpression = not implicitYieldEnabled
 
         match comp with
-        | SynExpr.New _ ->
```

Comment:
As we are rewriting `{ new Foo() }` as `SynExpr.ObjExpr` we no longer need this ad-hoc check here.